lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -340,3 +345,63 @@ def dump_grading_context(course):
     msg += "length=%d\n" % len(gcontext['all_descriptors'])
     msg = '<pre>%s</pre>' % msg.replace('<', '&lt;')
     return msg
+
+
+def iterate_problem_components(course):
+    """ Helper function for iterating through all problems of a course """
```

Comment:
s/function/generator/

otherwise this looks real scary... until you get to the `yield` :)